src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -607,6 +609,21 @@ void TensorIterator::executeDynamicImpl(dnnl::stream strm) {
     bool continue_cond = initial_cond_check->getStatus();
     int max_num_iter = trip_count_check->getStatus();
 
+    // redefine memory for input nodes of subgraph
+    if (checkForInputAndBodyShapesInequality()) {
+        reshapeSubgraphInput();
+
+        first_mappers.clear();
+        const auto &eng = getEngine();
+        for (auto map_rule : inputPortMap) {
+            auto from_mem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
+            auto &to_mem = input_mems[map_rule.to].front();  // first memory is enough to access the shared underlying physical memory
+
+            if (map_rule.axis == -1)
+                first_mappers.emplace_back(std::make_shared<BackEdgePortHelper>(context->getParamsCache(), from_mem, to_mem, eng));
+        }
+    }
```

Comment:
Is it possible to move this code to the `prepareParams` stage to avoid extra overhead on the `exec` call?

Reply:
That's the problem - the TI node is static while the subgraph is dynamic and backedge at the end of each iteration changes the subgraph input memory. 
So prepareParams won't be called. And the else branch won't take place in this situation.

Reply:
>prepareParams won't be called for static node. And the else branch won't take place in this situation.
  It's called for static node, if `needPrepareParams()` is true. However, if it's false for static shapes (or at least for the particular case caused the issue), there is a special condition `runAsDynamic()` designed specially for such scenarios. So all the necessary preparation steps may be moved under this condition check, please refer https://github.com/openvinotoolkit/openvino/blob/ca5a69eea01bee4fc0011bd034027f5a545bbace/src/plugins/intel_cpu/src/nodes/tensoriterator.cpp#L514-L515

Reply:
updated as we discussed online